### DISCUSSION TITLE
Hybrid RC and GC

### DISCUSSION THREAD

**Discussion by nordlow** [Neat-Lang/neat#30] [Ideas]
Have you looked into V's memory model? If not it might be of interest to you.

Have you thought about having an optional manual GC pass that can collect RC cycles? IRC, that's what V does.

**Comment by FeepingCreature** [comment]
If you preload Boehm, that's exactly what you get! :)

Personally, I simply haven't run into any cycles yet. I suspect the threat of cycles is overrated.

**Reply by nordlow** [reply]
Do you have any clues about what's making the Neat compiler slow as mentioned in your talk?

**Reply by FeepingCreature** [reply]
There's no one specific "slow thing". (Aside of LLVM.) There's some optimizations I could make to improve macro loading, but mostly it's just the slow morass of a hundred 1% fixes. Far as I can tell, anyway.

**Reply by FeepingCreature** [reply]
To be clear, with a warm LLVM cache, "slow" here is "competetive with D".
